codebasin/preprocessor.py

```diff
@@ -1352,8 +1352,13 @@ def parse(self):
             # warnings for common directives that shouldn't impact
             # correctness.
             common_unhandled = ["line", "warning", "error"]
-            if len(self.tokens) > 2 and str(self.tokens[1]) not in common_unhandled:
-                log.warning("Unrecognized directive")
+            if len(self.tokens) >= 2 and str(self.tokens[1]) not in common_unhandled:
+                filename = "<unknown>"
```

Comment:
Just wanted to make sure this is intended to be hardcoded? Seems like a pretty trivial thing to log...

Reply:
Good question.  This was deliberate, but you pointing it out encouraged me to take a second look at this.

The problem was that the piece of code that originally issued this warning is a generic `DirectiveParser` that is used in multiple places to tokenize strings.  It's not always associated with a file, and I couldn't think of a simple way to refactor things to track the filename some of the time.

What I've done in https://github.com/intel/code-base-investigator/pull/24/commits/08bf73b84d6cdb0b380bc8c031093783825ccb5e is move the warning out of the preprocessor and into the parser (which _always_ know the filename).  The parser probably should have been responsible for issuing the warning in the first place, so thank you for pointing this out.